### PR TITLE
Check for custom exokit:// urls

### PR DESCRIPTION
**Pull request by chrisplatorres**
This PR proposes adding to the current window scripts a check for the custom `exokit://` URLs.

### CODE REVIEW

**Comment by modulesio** [comment]
If the intent is for Exokit to support the `exokit://` URL scheme, can we just do that in `index.js`?

Or is there a good reason the windows start shell script needs to be involved here?

**Comment by chrisplatorres** [comment]
This was put together in a very short-sighted/"don't worry be hacky" way, will start over and do it in `index.js` instead.

File: src/index.js [hunk 1]
```diff
@@ -1059,7 +1059,16 @@ const _start = () => {
...
+    if (u.includes("exokit://")){
+      u = u.replace(/^exokit:\/\//, '');
+    }
+    if (u.includes("https//") || u.includes("http//")){
+      u = u.replace(/(^\w+|^)\/\//, '');
+    }
+    if (u.includes("https://") || u.includes("http://")){
+      u = u.replace(/(^\w+:|^)\/\//, '');
+    }
+    u = 'http://' + u;
```
Comment (chrisplatorres) [inline src/index.js]: Should this be http:// or https:// by default?

**Comment by chrisplatorres** [comment]
Tested cases:

`exokit://https://aframe.io/a-painter`
`exokit://https//aframe.io/a-painter`
`https://aframe.io/a-painter`
`https//aframe.io/a-painter`
`exokit://aframe.io/a-painter`
`exokit://http://aframe.io/a-painter`
`exokit://http//aframe.io/a-painter`
`http://aframe.io/a-painter`
`http//aframe.io/a-painter`

File: src/index.js [hunk 2]
```diff
@@ -1059,7 +1059,16 @@ const _start = () => {
     if (u === '.') {
       console.warn('NOTE: You ran `exokit . <url>`\n(Did you mean to run `node . <url>` or `exokit <url>` instead?)')
     }
-    u = u.replace(/^exokit:/, '');
+    if (u.includes("exokit://")){
```
Comment (modulesio) [inline src/index.js]: Probably should be regexes. It's not unthinkable that a URL would be in a query string, which would throw all of this off.

File: src/index.js (hunk 1 above)
Comment (modulesio) [reply src/index.js]: `http://`

File: src/index.js [hunk 3]
```diff
@@ -1059,7 +1059,16 @@ const _start = () => {
...
+      u = u.replace(/^exokit:\/\//, '');
+    }
+    if (u.includes("https//") || u.includes("http//")){
```
Comment (modulesio) [inline src/index.js]: This would not be valid so I don't see the need to check it.

Comment (chrisplatorres) [reply src/index.js]: It is a case in discord when using `<exokit://https://test.com>` it turns it into `<exokit://https//test.com`.

File: src/index.js (hunk 2 above)
Comment (chrisplatorres) [reply src/index.js]: > It's not unthinkable that a URL would be in a query string

What would that look like? `someurl.com?h=exokit://othersite.com`

Comment (modulesio) [reply src/index.js]: More likely it would be `http://`, but yes.

File: src/index.js (hunk 3 above)
Comment (modulesio) [reply src/index.js]: If it's a bug in discord, we should fix it there rather than adding magic to the engine. Is it possible to fix on the Discord end?

Comment (chrisplatorres) [reply src/index.js]: Pending response from Discord.

**Comment by modulesio** [comment]
Actually -- shouldn't it be `exokit:` only? For protocol handlers I think that's how it's done. The `//` would be the start of the legitimate path.

So a correct exokit link would be `exokit:https://aframe.io/a-painter`

**Comment by chrisplatorres** [comment]
Both seem to work.

**Pull request closed without merging**